taas/mapping.py

```diff
@@ -0,0 +1,99 @@
+from abc import ABCMeta, abstractmethod
+
+
+class Mapping(object):
+    """
+        This class handles mappings between our raw input, and what we generate.
```

Comment:
It is! The triple-quote strings are all [docstrings](https://www.python.org/dev/peps/pep-0257/). I've been hoping someone (human or robot) would arrive telling me that I've been doing them wrong, but nobody has. They're apparently written in Restructured Text (RST), which I don't know, so I'm expecting the odd formatting error if we run a [documentation tool](https://wiki.python.org/moin/DocumentationTools) over them.

Having said that, I totally made an executive decision to have documentation which might contain formatting errors than to delay having documentation entirely until I'd learnt RST. :)